### ISSUE TITLE
"date" package is not using the option location

### ISSUE THREAD

**Issue by dsilletti** [influxdata/flux#4260]
I am using date.truncate() to query the data starting from the beginning of the month or the beginning of the day, in combination with option location with timezone.

The date.truncate() is still working in UTC. 

e.g.

```
import "date"
import "timezone"

option location = timezone.location(name: "Europe/Rome")

from(bucket: "hour")
    |> range(start: date.truncate(t: -2mo, unit: 1mo))

```

the result is starting from midnight UTC and not using the Europe/Rome timezone.

I have the same issue using other functions in the date package, such as:

- date.hour(t: r._time)
- date.monthDay(t: r._time)
- date.weekDay(t: r._time)
- date.month(t: r._time)

the return value is in UTC and not according the option location

**Comment by nathanielc** [comment]
DOD:

- Add support for `location` option to each of the date.* function where appropriate
- Add support for `location` option to any duration math functions i.e. experimental.addDuration experimental.subDuration

**Comment by dsilletti** [comment]
any update on this? thanks

**Comment by skartikey** [comment]
@dsilletti The PR is up and being reviewed by @jsternberg

**Comment by OdneOksavik** [comment]
Hi. Will this thread be updated when this fix is implemented in the latest stable release? If so I will subscribe to it, we have now added some hours to fix this issue for ourselves.

**Comment by nathanielc** [comment]
This issue was fixed in Flux version 0.149.0 see https://github.com/influxdata/flux/releases/tag/v0.149.0 

As for the next stable release of InfluxDB it will contain this fix, we have already merged this fix into that project. However this issue will not get an update when that release occurs. 

FWIW you can determine the Flux version you are using with this query https://github.com/influxdata/flux/blob/master/stdlib/runtime/runtime.flux#L12-L16

**Comment by OdneOksavik** [comment]
Thank you @nathanielc ! I can now see that we are running v.0.139.0.
Then I make a note of updating queries when we update InfluxDB.
